src/http_workers/work_runner.rs

```diff
@@ -94,11 +97,58 @@ where
         }
     }
 
+    async fn upload_once<U: AsRef<str>>(
+        &self,
+        twin: &Twin,
+        uri_path: U,
+        msg: &mut Message,
+        payload: UploadPayload,
+    ) -> Result<(), SendError> {
+        let mut mpart = MultipartRequest::default();
+
+        mpart.add_file("upload", payload.path);
+        let uri = uri_builder(uri_path, &twin.address);
+        log::debug!("sending upload to '{}'", uri);
+
+        let request = Request::post(uri)
+            .header(
+                header::CONTENT_TYPE,
+                format!("multipart/form-data; boundary={}", mpart.get_boundary()),
+            )
+            .header("rmb-upload-cmd", payload.cmd)
+            .header("rmb-source-id", payload.source)
+            .header("rmb-timestamp", payload.timestamp)
+            .header(
+                "rmb-signature",
+                payload.signature.unwrap_or_else(|| "".to_string()),
```

Comment:
I think this should be an error. if signature is not set. It need to be an error. 
I know that internally the get_upload_payload will fail if it failed to sign it but this code makes it unclear and also shows that if signature is not set it's possible to send an empty signature. which is invalid.

Instead, 
```rust
let payload = <get the payload someone> // check other comment on proper way to construct this
let signature = payload.sign()?; // that's one way to do it
```

A more abstract way to do it is to construct MultiPart request from the Payload. In general this code should look like

```rust
let mpart = payload.to_request(&identity)?; 
```
this then can implement and hide/abstract:
- what headers are set 
- how request is signed,
- etc ...